velox/connectors/hive/storage_adapters/abfs/AbfsFileSystem.cpp

```diff
@@ -60,7 +60,8 @@ class AbfsReadFile::Impl {
             connectStr, abfsAccount.fileSystem(), fileName_));
   }
 
-  void initialize() {
+  void initialize(const FileOptions& options) {
+    length_ = options.getFileSize();
```

Comment:
If the value `options.getFileSize()` is present but is ` < 0`, then we need to throw an error here since the expectation in this method is that `length_ >= 0`.

Reply:
Please add a test for this.